lib/rubocop/cop/performance/numeric_predicate.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Identifies places where numeric uses predicates `positive?`, and `negative?` should be
+      # converted to compare operator.
+      #
+      # @safety
+      #   This cop is unsafe because it cannot be guaranteed that the receiver
+      #   defines the predicates or can be compared to a number, which may lead
+      #   to a false positive for non-standard classes.
+      #
+      # @example
+      #   # bad
+      #   1.positive?
+      #   1.43.negative?
+      #   -4.zero?
+      #
+      #   # good
+      #   1 > 0
+      #   1.43 < 0.0
+      #   -4 == 0
+      #
+      class NumericPredicate < Base
+        extend AutoCorrector
+
+        MSG = 'Use compare operator `%<good>s` instead of `%<bad>s`.'
+        RESTRICT_ON_SEND = %i[positive? zero? negative?].freeze
+        REPLACEMENTS = { negative?: '<', positive?: '>', zero?: '==' }.freeze
+
+        def_node_matcher :num_predicate?, <<~PATTERN
+          (send $numeric_type? ${:negative? :positive? :zero?})
```

Comment:
Should this also look for [`nonzero?`](https://docs.ruby-lang.org/en/3.0/Numeric.html#method-i-nonzero-3F)